tests/rptest/tests/read_replica_e2e_test.py

```diff
@@ -74,6 +73,18 @@ def start_producer(self):
             message_validator=is_int_with_prefix)
         self.producer.start()
 
+    def create_read_replica_topic_success(self):
+        try:
+            self.create_read_replica_topic()
+        except RpkException as e:
+            if "The server experienced an unexpected error when processing the request" in str(
```

Comment:
I know this function is just getting moved, but it would benefit from a comment explaining why "unexpected error" is the _expected_ error in this context -- I'm guessing this is because read replica topic creation doesn't have a suitable kafka error code and so it uses UNKNOWN_SERVER_ERROR?

(Maybe this was already discussed, but perhaps something like UNKNOWN_TOPIC_OR_PARTITION would help to distinguish this case from true internal errors)

Reply:
cc @mattschumpert do you have an opinion on error code when redpanda can't create read replica topic because there's no corresponding data in S3?

Reply:
@LenaAn ,

something like

'Error creating Remote Read Replica topic: No topic data found for topic {foo} at s3://{bucket_url_with_pathl}'

Reply:
@mattschumpert : I think the request here is for suggestion of a good/relevant kafka error code.

Reply:
@piyushredpanda , @LenaAn 

In this case, I suggest:

LOG_DIR_NOT_FOUND | 57 | False | The user-specified log directory is not found in the broker config.

or

KAFKA_STORAGE_ERROR | 56 | True | Disk error when trying to access log file on the disk.


LOG_DIR_NOT_FOUND is actually quite accurate.